api/src/main/java/org/openmrs/module/aijar/dataintegrity/InvalidTBEncounters.java

```diff
@@ -0,0 +1,211 @@
+		ruleResults.addAll(singlePatientWithDuplicateTBNumberAcrossMultipleTreatmentPrograms(AijarConstants.DISTRICT_TB_NUMBER, "District TB Number"));
+		ruleResults.addAll(singlePatientWithDuplicateTBNumberAcrossMultipleTreatmentPrograms(AijarConstants.UNIT_TB_NUMBER, "Unit TB Number"));
+		ruleResults.addAll(singlePatientWithDuplicateTBNumberAcrossMultipleTreatmentPrograms(AijarConstants.HSD_TB_NUMBER, "HSD TB Number"));
+		ruleResults.addAll(multiplePatientsWithTheSameTBIdentifiers(AijarConstants.DISTRICT_TB_NUMBER, "District TB Number"));
+		ruleResults.addAll(multiplePatientsWithTheSameTBIdentifiers(AijarConstants.UNIT_TB_NUMBER, "Unit TB Number"));
+		ruleResults.addAll(multiplePatientsWithTheSameTBIdentifiers(AijarConstants.HSD_TB_NUMBER, "HSD TB Number"));
+		ruleResults.addAll(patientsWithNoFinalOutcomeNineMonthsAfterStartOfTreatment(DateTime.now()));
+		return ruleResults;
+	}
+
+	/**
+	 * Patients with similar TB identifiers: duplicated across multiple patients
+	 * @return List<RuleResult<Patient>>
+	 */
+	public List<RuleResult<Patient>> multiplePatientsWithTheSameTBIdentifiers(String identifierConceptUuid, String identifierTitle) {
+		log.info("Executing rule to find Patients with similar TB identifiers: duplicated across multiple patients");
+		String queryString = "SELECT o FROM Obs o WHERE o.voided = false AND o.encounter.patient.dead = 0 AND o.encounter.voided = 0 AND o.concept.uuid = :identifierConceptUuid AND o.valueText IN " 
+				+ " (SELECT ob.valueText FROM Obs ob WHERE ob.concept.uuid = :identifierConceptUuid AND ob.voided = 0 GROUP BY ob.concept.conceptId, ob.valueText HAVING COUNT(ob.valueText) > 1)"; 
+		Query query = getSession().createQuery(queryString);
+		query.setParameter("identifierConceptUuid", identifierConceptUuid);
+		
+		List<Obs> obsList = query.list();
+		Set<Patient> uniquePatientList = new HashSet<Patient>();
+		List<RuleResult<Patient>> ruleResults = new ArrayList<>();
+		for (Obs obs : obsList) {
+			Patient patient = obs.getEncounter().getPatient();
+			
+			RuleResult<Patient> ruleResult = new RuleResult<>();
+			ruleResult.setActionUrl("htmlformentryui/htmlform/editHtmlFormWithStandardUi.page?patientId=" + patient.getUuid() + "&encounterId=" + obs.getEncounter().getId());
+			ruleResult.setNotes("The " + identifierTitle + " " + getTbNumber(patient, obs.getEncounter(), identifierConceptUuid) + " is being used by another patient");
+			ruleResult.setEntity(patient);
+			
+			ruleResults.add(ruleResult);
+			uniquePatientList.add(patient);
+		}
+		log.info("There are " + uniquePatientList.size() + " Patients with similar " + identifierTitle);
+		
+		return ruleResults;
+	}
+
+	/**
+	 * Patients with similar TB identifiers: duplicated for a single patient,
+	 * @return List<RuleResult<Patient>>
+	 */
+	public List<RuleResult<Patient>> singlePatientWithDuplicateTBNumberAcrossMultipleTreatmentPrograms(String identifierConceptUuid, String identifierTitle) {
```

Comment:
@moshonk This method should have a filter for the same patient otherwise it duplicates the functionality of multiplePatientsWithTheSameTBIdentifiers. Also ensure that multiplePatientsWithTheSameTBIdentifiers does not return the same patient